### ISSUE TITLE
ImapFlow Fails to Retrieve Emails from Xeams Server

### ISSUE THREAD

**Issue by umanamente** [postalsys/imapflow#237]
**Describe the bug**
When connecting ImapFlow to a [Xeams Mail Server](https://www.xeams.com/home.htm), the client successfully authenticates and retrieves folders but does not return any email data. The same configuration works without issues when using an IMAP account on an Exchange Server or gmail. 

The Thunderbird client successfully retrieves emails from Xeams with the same configuration.

The ImapFlow client generates two empty warnings when reading two emails from the server and then returns an empty list of emails.

**To Reproduce**

1. Set up a Xeams mail server (**test environment credentials provided below**).
2. Configure ImapFlow to connect to the Xeams IMAP server with the following details:
   - Server: `144.24.250.201`
   - Email: `hello@world.com`
   - Password: `H8ll0W0rld`
   - IMAP Port: `143`
3. Attempt to fetch emails from the Xeams inbox.
4. Observe the following:
   - Folders are listed correctly.
   - Email metadata appears in server logs.
   - No email data is returned by ImapFlow.

**Expected behavior**
ImapFlow should return the email data retrieved from the Xeams server.

**Technical Details**
- **ImapFlow version**: v1.0.169
- **Mail server**: Xeams IMAP

**Log from Xeams server**

```
2024-11-14 15:26:09,243 -  - [      13] ************** New Connection from: 192.168.0.5:143 ************** 
2024-11-14 15:26:09,244 -  - [      13] S <-- * OK Synametrics IMAP4rev1 server ready 11/14/24 3:26 PM
2024-11-14 15:26:09,258 -  - [      13] C --> 1 CAPABILITY
2024-11-14 15:26:09,259 -  - [      13] S <-- * CAPABILITY IMAP4rev1 IDLE
2024-11-14 15:26:09,259 -  - [      13] S <-- 1 OK CAPABILITY completed
2024-11-14 15:26:09,306 -  - [      13] C --> 2 LOGIN "test@mailaccount.com" xxxxxx
2024-11-14 15:26:09,307 -  - [      13] S <-- 2 OK LOGIN completed
2024-11-14 15:26:09,329 -  - [      13] C --> 3 CAPABILITY
2024-11-14 15:26:09,329 -  - [      13] S <-- * CAPABILITY IMAP4rev1 IDLE
2024-11-14 15:26:09,329 -  - [      13] S <-- 3 OK CAPABILITY completed
2024-11-14 15:26:09,365 -  - [      13] C --> 4 LIST "" ""
2024-11-14 15:26:09,365 -  - [      13] S <-- * LIST (\Noselect) "/" ""
2024-11-14 15:26:09,365 -  - [      13] S <-- 4 OK LIST completed
2024-11-14 15:26:09,397 -  - [      13] C --> 5 LIST "" "INBOX"
2024-11-14 15:26:09,400 -  - [      13] S <-- * LIST () "/" "INBOX"
2024-11-14 15:26:09,400 -  - [      13] S <-- 5 OK LIST completed
2024-11-14 15:26:09,424 -  - [      13] C --> 6 LSUB "" "INBOX"
2024-11-14 15:26:09,425 -  - [      13] S <-- * LSUB () "/" "INBOX"
2024-11-14 15:26:09,425 -  - [      13] S <-- 6 OK LIST completed
2024-11-14 15:26:09,436 -  - [      13] C --> 7 EXAMINE INBOX
2024-11-14 15:26:09,454 -  - [      13] S <-- * FLAGS (\Seen \Deleted)
2024-11-14 15:26:09,454 -  - [      13] S <-- * 2 EXISTS
2024-11-14 15:26:09,454 -  - [      13] S <-- * OK [UIDNEXT 3] Predicted next UID
2024-11-14 15:26:09,454 -  - [      13] S <-- * 0 RECENT
2024-11-14 15:26:09,454 -  - [      13] S <-- * OK [PERMANENTFLAGS (\Deleted \Seen)]
2024-11-14 15:26:09,454 -  - [      13] S <-- 7 OK [READ-ONLY] SELECT completed
2024-11-14 15:26:09,496 -  - [      13] C --> 8 UID SEARCH ALL
2024-11-14 15:26:09,499 -  - [      13] S <-- * SEARCH 1 2
2024-11-14 15:26:09,506 -  - [      13] S <-- 8 OK UID completed
2024-11-14 15:26:09,517 -  - [      13] C --> 9 UID FETCH 1:2 (UID ENVELOPE)
2024-11-14 15:26:09,789 -  - [      13] S <-- * 1 FETCH (UID 1 ENVELOPE ("Thu, 14 Nov 2024 12:30:02 +0000" "<SUBJECT>" (("hello@world.com" NIL "hello" "world.com")) (("hello@world.com" NIL "hello" "world.com")) (("hello@world.com" NIL "hello" "world.com")) (("helloFROM@worldfrom.com" NIL "helloFROM" "worldfrom.com")) NIL NIL NIL "<ZR0P278MB02686EC0E225A0AAA9DB847FC95B2@ZR0P278MB0268.CHEP278.PROD.OUTLOOK.COM>"))
2024-11-14 15:26:09,812 -  - [      13] S <-- * 2 FETCH (UID 2 ENVELOPE ("Thu, 14 Nov 2024 12:04:05 +0000" "test" (("hello@world.com" NIL "hello" "world.com")) (("hello@world.com" NIL "hello" "world.com")) (("hello@world.com" NIL "hello" "world.com")) (("helloFROM@worldfrom.com" NIL "helloFROM" "worldfrom.com")) NIL NIL NIL "<ZR0P278MB0268AC40E3BF0386AED3C53CC95B2@ZR0P278MB0268.CHEP278.PROD.OUTLOOK.COM>"))
2024-11-14 15:26:09,812 -  - [      13] S <-- 9 OK UID FETCH completed
2024-11-14 15:26:09,864 -  - [      13] C --> A LOGOUT
2024-11-14 15:26:09,864 -  - [      13] S <-- * BYE IMAP4rev1 IDLE Server logging out
2024-11-14 15:26:09,864 -  - [      13] S <-- A OK LOGOUT completed
2024-11-14 15:26:09,865 -  - [      13] ~~~~~~~~~~~~~~ Connection closed: 192.168.0.5 (622 ms)~~~~~~~~~~~~~~ 

```

**Comment by andris9** [comment]
Could you elaborate a bit on what exactly does not work? Could you provide a testable code snippet I could run and verify that the behavior is not correct? I did some quick tests of my own, and the returned results seemed fine, so there must be something I'm missing.

**Comment by umanamente** [comment]
I noticed that the problem occurs only when using the `readOnly` option with `mailboxOpen`:  
```typescript
await client.mailboxOpen('INBOX', { readOnly: true });
```

Here’s the full code:
```typescript
const { ImapFlow } = require('imapflow');

const client = new ImapFlow({
    host: '144.24.250.201',
    port: 143,
    secure: false,
    auth: {
        user: 'hello@world.com',
        pass: 'H8ll0W0rld'
    }
});

(async () => {
    try {
        console.log('Connecting to IMAP server...');
        await client.connect();
        console.log('Connected!');

        console.log('Locking the mailbox...');
        await client.mailboxOpen('INBOX', { readOnly: true });
        
        console.log('Fetching emails...');
        for await (let message of client.fetch('1:*', { envelope: true, uid: true })) {
            console.log(``);
            console.log(`UID: ${message.uid}`);
            console.log(`From: ${message.envelope.from.map(f => f.address).join(', ')}`);
            console.log(`Subject: ${message.envelope.subject}`);
            console.log(``);
        }

        console.log('Closing the mailbox...');
        await client.mailboxClose();

    } catch (err) {
        console.error('Error:', err);
    } finally {
        await client.logout();
        console.log('Disconnected from IMAP server.');
    }
})();
```

**Comment by andris9** [comment]
It appears that the problem occurs when opening a mailbox in read-only mode using the `EXAMINE` command. The Xeams Mail Server does not return the `UIDVALIDITY` value in response to `EXAMINE`, which is a protocol violation. According to the IMAP specification, the response for `EXAMINE` must be exactly the same as for `SELECT`, including all mandatory data. See [RFC 3501 Section 6.3.2](https://www.rfc-editor.org/rfc/rfc3501.html#section-6.3.2):

> The EXAMINE command is identical to SELECT and returns the same output;

I will add an additional check in ImapFlow to handle cases where `UIDVALIDITY` is not returned. This should resolve the issue on the ImapFlow side.

**Comment by andris9** [comment]
Fixed in v1.0.170
